plugins/catalog-node/api-report-alpha.md

```diff
@@ -20,8 +21,8 @@ import { Validators } from '@backstage/catalog-model';
 
 // @alpha (undocumented)
 export interface CatalogAnalysisExtensionPoint {
-  // (undocumented)
-  addLocationAnalyzer(analyzer: ScmLocationAnalyzer): void;
+  addScmLocationAnalyzer(analyzer: ScmLocationAnalyzer): void;
```

Comment:
I was hoping to not have this at all, but instead opting for

```ts
setLocationAnalyzer(
  DefaultLocationAnalyzer.create({
    deps,
    analyzers: [new CustomThing()],
 })
);
```

Or similar and heading toward that as a general pattern (eg for the catalog model etc)

This odd duality of "add child - or replace parent and then children are ignored" would be nice to get rid of.

Reply:
This lets builders ship support for analysis of individual providers that can be installed without manual module wiring. We could ship the default one with a GitHub analyzer from the GitHub module but then you can't combine support form multiple SCMs without code changes.

Reply:
How do you mean?

Reply:
`setLocationAnalyzer((scmAnalyzers: ScmLocationAnalyzer[]) => Location Analyzer): void` so to speak.

Maybe I'm misunderstanding the interfaces somewhere and should wait until I get a chance to check the branch out locally and poke at it :)

Reply:
Ah yep I guess we could do that. Ofc there's no guarantee that the `LocationAnalyzer` implementation can or will use the SCM analyzers, but feels like an improvement still.

Reply:
Mkay updated with a new API for `setLocationAnalyzer`